source/lac/sparsity_tools.cc

```diff
@@ -183,7 +183,12 @@ namespace SparsityTools
       Assert ( numEdges != nullptr , ExcInternalError() );
 
       for (int i=0; i<num_obj; ++i)
-        numEdges[i] = graph->row_length(globalID[i])-1;
+        {
+          if ( graph->exists(i,i) )     //Check if diagonal element is present
```

Comment:
so does the new function also work in case user have diagonal entries at the sparsity pattern? That should not matter for coloring, of course.

Maybe add additional test (a copy of the present one) with the only difference being diagonal entries in sparsity pattern.

Reply:
Yes, the new function works even if we have diagonal entries. In fact, the sparsity pattern already has diagonal element by default even if I didn't add diagonal elements to dynamic sparsity pattern (before copying it back to sparsity pattern).

Reply:
Should I still add an additional test with no diagonal elements? Currently, sparsity pattern automatically adds diagonal elements when copied from dynamic sparsity pattern. If the test is needed i have to find some other way.

Reply:
> Should I still add an additional test with no diagonal elements?

no, i think it's fine.